src/client/components/map/Map.tsx

```diff
@@ -159,12 +159,18 @@ const Map = ({
 
   // Update labels when selection is changed
   useEffect(() => {
-    const visibility = label === undefined ? "none" : "visible";
-    // TODO: hardcoding county because we can't set the geolevel yet. This
-    // should instead display only for the current geolevel (GH#200)
-    map && map.setLayoutProperty("county-label", "visibility", visibility);
-    map && map.setLayoutProperty("county-label", "text-field", `{${label}}`);
-  }, [map, label]);
+    map &&
+      staticMetadata.geoLevelHierarchy.forEach(geoLevel =>
+        map.setLayoutProperty(`${geoLevel.id}-label`, "visibility", "none")
+      );
+    map && map.setLayoutProperty(`${selectedGeolevel.id}-label`, "visibility", "visible");
+    map &&
+      map.setLayoutProperty(
+        `${selectedGeolevel.id}-label`,
+        "text-field",
+        label ? `{${label}}` : ""
+      );
+  }, [map, label, geoLevelIndex]);
```

Comment:
One small change here to make the warnings go away:

```suggestion
  }, [map, label, staticMetadata, selectedGeolevel]);
```